### PR TITLE
Add CI based on zlib-rs

### PR DESCRIPTION
**Pull request by bjorn3**
Also fix a bunch of portability issues.

### CODE REVIEW

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@56ab7930c591507f833cbaed864d201386d518a8
+        with:
+          tool: cargo-nextest
+      - name: Test public C api with NULL arguments
+        run: "cargo +nightly miri nextest run -j4 -p test-libbzip2-rs-sys null::"
+        env:
+          RUSTFLAGS: "-Ctarget-feature=+avx2"
+      - name: Test allocator with miri
+        run: "cargo +nightly miri nextest run -j4 -p libbzip2-rs-sys allocate::"
```
Comment (folkertdev) [inline .github/workflows/checks.yaml:374]: this matches no tests in our case right? we should make a subset somehow that can run in < 1min

File: Cargo.toml [hunk 2]
```diff
@@ -12,6 +12,7 @@ authors = ["C2Rust"]
 version = "0.0.0"
 publish = false
 edition = "2021"
+rust-version = "1.82" # MSRV
```
Comment (folkertdev) [inline Cargo.toml:15]: do we need this? for usage in the `bzip2` crate it might be nice to have a substantially lower msrv?

Comment (bjorn3) [reply Cargo.toml:15]: The CI config expects some MSRV. I just picked the latest rustc as that is what I'm locally developing on. The actual MSRV is at least 1.77 though for the binaries as I used `c"foo"` C string literals.

File: .github/workflows/checks.yaml (hunk 1 above)
Comment (bjorn3) [reply .github/workflows/checks.yaml:374]: Right...

File: Cargo.toml (hunk 2 above)
Comment (bjorn3) [reply Cargo.toml:15]: We can always rewrite code that uses newer features again in the future if necessary.

Comment (folkertdev) [reply Cargo.toml:15]: the `c"foo"` literals can mostly go as we clean up the code I think? at least that has been the case in `bzlib.rs`. but I'm fine with picking this now and lowering it later when inclusion in `bzip2` becomes more concrete.

**Pull request merged**